src/platform/packages/shared/kbn-ebt-tools/src/performance_tracker.ts

```diff
@@ -0,0 +1,140 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import { groupBy } from 'lodash';
+
+import type { Logger } from '@kbn/logging';
+
+/**
+ * PERFORMANCE_TRACKER_TYPES defines top-level categories to be used as
+ * the mark name. They are used to group marks and measures by type.
+ */
+export const PERFORMANCE_TRACKER_TYPES = {
+  PANEL: 'Panel',
+} as const;
+export type PerformanceTrackerTypes =
+  (typeof PERFORMANCE_TRACKER_TYPES)[keyof typeof PERFORMANCE_TRACKER_TYPES];
+
+/**
+ * PerformanceTrackerMarks are the marks that can be used to track performance
+ * of lens charts. They are used to mark specific points in time during
+ * the chart's lifecycle.
+ */
+export const PERFORMANCE_TRACKER_MARKS = {
+  /**
+   * Mark that indicates the start of everything before the rendering process.
+   */
+  PRE_RENDER: 'preRender',
+  /**
+   * Mark that indicates the start of the rendering process.
+   * Should be used right before returning the chart's JSX.
+   */
+  RENDER_START: 'renderStart',
+  /**
+   * Mark that indicates the end of the rendering process.
+   * Should be used at the beginning of the `renderComplete` callback.
+   */
+  RENDER_COMPLETE: 'renderComplete',
+} as const;
+export type PerformanceTrackerMarks =
+  (typeof PERFORMANCE_TRACKER_MARKS)[keyof typeof PERFORMANCE_TRACKER_MARKS];
+
+export const PERFORMANCE_TRACKER_MEASURES = {
+  PRE_RENDER_DURATION: 'preRenderDuration',
+  RENDER_DURATION: 'renderDuration',
+} as const;
+export type PerformanceTrackerMeasures =
+  (typeof PERFORMANCE_TRACKER_MEASURES)[keyof typeof PERFORMANCE_TRACKER_MEASURES];
+
+/**
+ * Options to create a performance tracker.
+ */
+interface PerformanceTrackerOptions {
+  /**
+   * High-level type of the performance tracker, for example "Panel".
+   */
+  type: PerformanceTrackerTypes;
+  /**
+   * Instance of the performance tracker type, for example "xyVis".
+   * This is used to group marks and measures by instance.
+   */
+  instance: string;
```

Comment:
"instance" does not see like the right word. To me an instance would be a specific panel on a dashboard. How about a word that is less overloaded, like "subType"?

Reply:
👍  - renamed in c1274919a09.